ui/src/RecentFilesDrawer/AddTagToWorkflowPopover.tsx

```diff
@@ -130,15 +136,19 @@ export default function AddTagToWorkflowPopover({ workflow }: Props) {
               size={"xs"}
               px={5}
               isDisabled={newTagName.length === 0}
-              onClick={async () => {
-                await tagsTable?.upsert(newTagName);
-                tagsTable?.listAll().then((tags) => setAllTags(tags ?? []));
-                setNewTagName("");
-              }}
+              onClick={addTag}
             >
               New Tag
             </Button>
           </HStack>
+          {submitError ? (
+            <Alert status="error" size="small">
+              <AlertIcon />
+              The name is duplicated, please modify it and submit again.
```

Comment:
we can just use tagsTable.put() and don't check for duplication

Reply:
done